### ISSUE TITLE
Failing SSL connections

### ISSUE THREAD

**Issue by dkocher** [DICE-UNC/jargon#223]
Running the current mater branch yields when connecting to _data.iplantcollaborative.org_.

`org.irods.jargon.core.exception.JargonException: unable to start SSL socket`

**Comment by michael-conway** [comment]
Let's talk w Dennis this morning this is likely server config.

**Comment by dkocher** [comment]
Possibly not as it works using the latest release version.

**Comment by michael-conway** [comment]
OK good data point, yes, we're into field testing on SSL and there are just a few edge cases remaining...that is, combinations of PAM and some SSL negotiation permutations. Those work, but there are some things showing up in the logs to clean up.

**Comment by dkocher** [comment]
We still have the same failure.

```
2017-01-24 14:11:51,588 [main] ERROR org.irods.jargon.core.connection.SslConnectionUtilities - ssl exception in handshake
javax.net.ssl.SSLHandshakeException: Remote host closed connection during handshake
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:992)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1403)
	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1387)
	at org.irods.jargon.core.connection.SslConnectionUtilities.createSslSocketForProtocol(SslConnectionUtilities.java:166)
	at org.irods.jargon.core.connection.SslConnectionUtilities.createSslSocketForProtocolAndIntegrateIntoProtocol(SslConnectionUtilities.java:200)
	at org.irods.jargon.core.connection.ClientServerNegotiationService.wrapConnectionInSslIfConfigured(ClientServerNegotiationService.java:210)
	at org.irods.jargon.core.connection.ClientServerNegotiationService.notifyServerOfNegotiationSuccess(ClientServerNegotiationService.java:182)
	at org.irods.jargon.core.connection.ClientServerNegotiationService.negotiateUsingServerProtocol(ClientServerNegotiationService.java:142)
	at org.irods.jargon.core.connection.ClientServerNegotiationService.negotiate(ClientServerNegotiationService.java:111)
	at org.irods.jargon.core.connection.AuthMechanism.clientServerNegotiation(AuthMechanism.java:174)
	at org.irods.jargon.core.connection.AuthMechanism.clientServerNegotiationHook(AuthMechanism.java:78)
	at org.irods.jargon.core.connection.AuthMechanism.authenticate(AuthMechanism.java:214)
	at org.irods.jargon.core.connection.AbstractIRODSMidLevelProtocolFactory.authenticate(AbstractIRODSMidLevelProtocolFactory.java:241)
	at org.irods.jargon.core.connection.AbstractIRODSMidLevelProtocolFactory.instance(AbstractIRODSMidLevelProtocolFactory.java:103)
	at org.irods.jargon.core.connection.IRODSProtocolManager.createNewProtocol(IRODSProtocolManager.java:146)
	at org.irods.jargon.core.connection.IRODSSimpleProtocolManager.getIRODSProtocol(IRODSSimpleProtocolManager.java:69)
	at org.irods.jargon.core.connection.IRODSSession.connect(IRODSSession.java:464)
	at org.irods.jargon.core.connection.IRODSSession.currentConnection(IRODSSession.java:371)
	at org.irods.jargon.core.pub.IRODSGenericAO.<init>(IRODSGenericAO.java:65)
	at org.irods.jargon.core.pub.IRODSFileSystemAOImpl.<init>(IRODSFileSystemAOImpl.java:74)
	at org.irods.jargon.core.pub.IRODSAccessObjectFactoryImpl.getIRODSFileSystemAO(IRODSAccessObjectFactoryImpl.java:356)
	at ch.cyberduck.core.irods.IRODSSession.connect(IRODSSession.java:85)
	at ch.cyberduck.core.irods.IRODSSession.connect(IRODSSession.java:66)
	at ch.cyberduck.core.Session.open(Session.java:135)
	at ch.cyberduck.core.irods.IRODSAttributesFinderFeatureTest.testFind(IRODSAttributesFinderFeatureTest.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.io.EOFException: SSL peer shut down incorrectly
	at sun.security.ssl.InputRecord.read(InputRecord.java:505)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	... 53 more
```

**Comment by michael-conway** [comment]
To CyVerse? I'll talk to Tony and Dennis today....

**Comment by dkocher** [comment]
Connecting to `data.iplantcollaborative.org`.

**Comment by michael-conway** [comment]
sure, we'll test from the base unit tests to help clarify

**Comment by dkocher** [comment]
Has this been resolved in 4.1.10.0-RELEASE?
